Library/Homebrew/diagnostic.rb

```diff
@@ -703,7 +703,7 @@ def check_git_status
             If this is a surprise to you, then you should stash these modifications.
             Stashing returns Homebrew to a pristine state but can be undone
             should you later need to do so for some reason.
-              cd #{path} && git stash -u && git clean -d -f
+              git stash -u -C #{path} && git clean -d -f -C #{path}
```

Comment:
```suggestion
              git -C "#{path}" stash -u && git -C "${path}" clean -d -f
```
`-C` should come before the subcommand:
```
❯ git stash -u -C /opt/homebrew
fatal: not a git repository (or any of the parent directories): .git

❯ git -C /opt/homebrew stash -u
Saved working directory and index state WIP on (no branch): a729c533fd6e1 Merge pull request #20377 from Homebrew/fix-linkage-checker-file-descriptors

```